types/midi/midi-tests.ts

```diff
@@ -0,0 +1,15 @@
+import midi from 'midi';
+
+const input = new midi.Input();
+input.getPortCount();
+input.getPortName(0);
+input.on('message', (deltaTime: number, message: number[]) => {});
+input.openPort(0);
+input.ignoreTypes(false, false, false);
+
+const output = new midi.Output();
+output.getPortCount();
+output.getPortName(0);
+output.openPort(0);
+output.sendMessage([176, 22, 1]);
+output.closePort();
```

Comment:
update your tests to

```suggestion
import midi = require('midi');
import fs = require('fs');

const input = new midi.Input();
input.getPortCount();
input.getPortName(0);
input.on('message', (deltaTime, message) => {
    deltaTime; // $ExpectType number
    message; // $ExpectType Message
});
input.openPort(0);
input.ignoreTypes(false, false, false);
input.closePort();
input.openVirtualPort('Test Input');

const output = new midi.Output();
output.getPortCount();
output.getPortName(0);
output.openPort(0);
output.sendMessage([176, 22, 1]);
output.closePort();

{
    // create a readable stream
    const stream1 = midi.createReadStream();

    // createReadStream also accepts an optional `input` param
    const stream2 = midi.createReadStream(input);

    stream2.pipe(fs.createWriteStream('something.bin'));
}

{
    // create a writable stream
    const stream1 = midi.createWriteStream();

    // createWriteStream also accepts an optional `output` param
    const stream2 = midi.createWriteStream(output);

    fs.createReadStream('something.bin').pipe(stream2);
}
```